web-api/src/business/useCases/saveCaseDetailInternalEditInteractor.ts

```diff
@@ -144,18 +146,30 @@ export const saveCaseDetailInternalEdit = async (
   } else {
     const petitionDocketEntry = caseEntity.getPetitionDocketEntry();
 
-    const initializeCaseWorkItem = petitionDocketEntry.workItem;
+    if (!petitionDocketEntry) {
+      throw new NotFoundError(
+        `Could not find petition docket entry on case ${petitionDocketEntry}`,
+      );
+    }
 
-    const workItemEntity = new WorkItem({
-      ...initializeCaseWorkItem,
-      assigneeId: user.userId,
-      assigneeName: user.name,
-      inProgress: true,
+    const petitionWorkItem = await getWorkItemByDocketNumberAndDocketEntryId({
+      docketNumber,
+      docketEntryId: petitionDocketEntry?.docketEntryId,
     });
 
-    await upsertWorkItems({
-      workItems: [workItemEntity.validate().toRawObject()],
-    });
+    if (petitionWorkItem) {
```

Comment:
Should we throw an error if no petitionWorkItem exists?

Reply:
I will look into this tomorrow.

Reply:
Narrator: He didn't.

Looking into now.

Reply:
Yes, we should. Thank you!